File: TwoSum/js/solution2.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    if (nums.length < 2) {
        return null;
    }
    let numMap = new Map();
    for (let i in nums) {
        if (numMap.has(nums[i])) {
            return [numMap.get(nums[i]), i];
        } else {
            numMap.set(target - nums[i], i);
        }
    }
    return null;
};

File: two-sum/java/solution2.java
import java.util.HashMap;

public class TwoSum {

    public static int[] findTwoSum(int[] nums, int target) {
        if (nums.length < 2) {
            return null;
        }

        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(nums[i])) {
                return new int[]{map.get(nums[i]), i};
            } else {
                map.put(target - nums[i], i);
            }
        }
        return null;
    }
}